src/data_structure/binary_tree.rs

```diff
@@ -0,0 +1,66 @@
+#[derive(PartialEq)]
+pub struct Node<'a> {
+    val: &'a str,
```

Comment:
Let's make this generic so that can can have anything as a value

Reply:
Do you think it's better to name it as `BinaryNode` since it has at most two children?